ste/xfer-deleteBlob.go

```diff
@@ -43,12 +43,14 @@ func DeleteBlobPrologue(jptm IJobPartTransferMgr, p pipeline.Pipeline, pacer pac
 		jptm.ReportTransferDone()
 	}
 
-	_, err := srcBlobURL.Delete(jptm.Context(), azblob.DeleteSnapshotsOptionNone, azblob.BlobAccessConditions{})
+	// TODO confirm whether we should add a new flag to allow user to specify whether snapshots should be removed
+	_, err := srcBlobURL.Delete(jptm.Context(), azblob.DeleteSnapshotsOptionInclude, azblob.BlobAccessConditions{})
 	if err != nil {
 		// If the delete failed with err 404, i.e resource not found, then mark the transfer as success.
 		if strErr, ok := err.(azblob.StorageError); ok {
 			if strErr.Response().StatusCode == http.StatusNotFound {
 				transferDone(common.ETransferStatus.Success(), nil)
+				return
```

Comment:
Consider logging a warning that sometimes 404s can be caused by failed authentication.

Reply:
404 is only given if no SAS/Oauth token is present. Either one of them had to be present for us to scan the resource.